klippa/src/hvar.rs

```diff
@@ -0,0 +1,640 @@
+                None => DeltaSetIndex {
+                    outer: (old_gid >> 16) as u16,
+                    inner: (old_gid & 0xFFFF) as u16,
+                },
+            };
+
+            let outer = v.outer;
+            if outer as usize >= inner_maps.len() {
+                continue;
+            }
+
+            let new_outer = outer_map.get(outer as u32).unwrap();
+            let new_inner = inner_maps[outer as usize].get(v.inner as u32).unwrap();
+            self.output_map
+                .insert(*new_gid, (*new_outer << 16) | *new_inner);
+        }
+    }
+
+    fn is_identity(&self) -> bool {
+        self.output_map.is_empty()
+    }
+
+    fn to_serialize_plan(&self) -> DeltaSetIndexMapSerializePlan {
+        DeltaSetIndexMapSerializePlan::new(
+            self.outer_bit_count,
+            self.inner_bit_count,
+            &self.output_map,
+            self.map_count as u32,
+        )
+    }
+}
+
+#[derive(Default)]
+pub(crate) struct HvarVvarSubsetPlan {
+    inner_maps: Vec<IncBiMap>,
+    index_map_subset_plans: Vec<IndexMapSubsetPlan>,
+}
+
+impl HvarVvarSubsetPlan {
+    pub(crate) fn new(
+        plan: &Plan,
+        var_store: &ItemVariationStore,
+        index_maps: &[Option<DeltaSetIndexMap>],
+    ) -> Result<Self, ReadError> {
+        let mut this = HvarVvarSubsetPlan::default();
+        let vardata_count = var_store.item_variation_data_count() as usize;
+
+        let mut inner_sets = Vec::new();
+        inner_sets.resize(vardata_count, Default::default());
+
+        let mut outer_map = IncBiMap::default();
+        // don't bypass empty adv map
+        let index_map_subset_plan =
+            IndexMapSubsetPlan::init(&index_maps[0], plan, false, &mut outer_map, &mut inner_sets)?;
+        this.index_map_subset_plans.push(index_map_subset_plan);
+
+        let mut adv_set = IntSet::empty();
+        if index_maps[0].is_none() {
+            adv_set.union(&inner_sets[0]);
+        }
+
+        for index_map in index_maps.iter().skip(1) {
+            this.index_map_subset_plans.push(IndexMapSubsetPlan::init(
+                index_map,
+                plan,
+                true,
+                &mut outer_map,
+                &mut inner_sets,
+            )?);
+        }
+
+        outer_map.sort();
+
+        let retain_adv_map = index_maps[0].is_none()
+            && plan
+                .subset_flags
+                .contains(SubsetFlags::SUBSET_FLAGS_RETAIN_GIDS);
+
+        if retain_adv_map {
+            let mut inner_map = IncBiMap::with_capacity(plan.new_to_old_gid_list.len());
+            for (_, old_gid) in plan.new_to_old_gid_list.iter() {
+                inner_map.add(old_gid.to_u32());
+            }
+            this.inner_maps.push(inner_map);
+        } else {
+            let mut inner_map = IncBiMap::default();
+            for old_gid in adv_set.iter() {
+                inner_map.add(old_gid as u32);
+            }
```

Comment:
same story. In general if there's a way to express something that avoids the`mut` keyword, you should try to do that, since it makes the code easier to read (every `mut` declaration is something that the reader needs to keep track of, in case it is subsequently modified)